src/DependencyInjection/SymfonycastsSassExtension.php

```diff
@@ -61,7 +61,7 @@ public function getConfigTreeBuilder(): TreeBuilder
                         ->end()
                     ->validate()
                         ->ifTrue(static function (array $paths): bool {
-                            if (1 === \count($paths)) {
+                            if (1 === \count($paths) || !array_is_list($paths)) {
```

Comment:
If you want to allow using maps and not just lists, you need to use `->useAttributeAsKey`. Otherwise, merging multiple config files (note that for this matter, a `when@prod` section counts as a separate config for the merging process) together won't preserve keys (and devs using a XML config file would not be able to specify a key)

Reply:
Not sure how I would define that?

```php
$rootNode
    ->children()
        ->arrayNode('root_sass')
            ->info('Path to your Sass root file')
            ->cannotBeEmpty()
            ->useAttributeAsKey('???')  // <--- here?
            ->scalarPrototype()
                ->end()
            ->validate()
                ...
```

With or without `->useAttributeAsKey()`, and manually testing it with an extra `when@dev` config for example, does not change the array passed to the `validate`-callback function? And what would be the `$name` variable to pass to `->useAttributeAsKey()`?

Can you point me to an example definition?

Reply:
@evertharmeling you would have to define a `when@dev` section (or a separate file) that also configures this `root_sass` node, so that some merging actually happens (if the value is defined only in 1 place, the merging is bypassed entirely, which will preserve keys even if the node is not configured for that).

For the name being passed, this would be the name of the attribute used in an XML config (where it cannot be an array key). In this case, `output` looks like a good option.

Reply:
Testing with:

`config/packages/symfony_casts.yaml`:
```yaml
symfonycasts_sass:
    root_sass:
        website: 'assets/website/scss/app.scss'
```

`config/packages/test.yaml`:
```yaml
symfonycasts_sass:
    root_sass:
        test: 'assets/website/scss/app.scss'
        website: 'assets/test/scss/app.scss'

when@dev:
    symfonycasts_sass:
        root_sass:
            test2: 'assets/website/scss/app.scss'
```

Running `symfony console debug:config SymfonycastsSassBundle`:
```bash
Current configuration for "SymfonycastsSassBundle"
==================================================

symfonycasts_sass:
    root_sass:
        website: assets/test/scss/app.scss
        test: assets/website/scss/app.scss
        test2: assets/website/scss/app.scss
    binary: null
    embed_sourcemap: true
```

Even replacing an entry with same key `website` seems to work? 

I'm fine with adding `->useAttributeAsKey('output')`, but can't reproduce it with these examples or am I still overlooking something?